### PR TITLE
FIX: distributed knn double sqrt bug

### PR DESCRIPTION
**Pull request by ethanglaser**
# Description
Removes unnecessary duplicate sqrt of distances
This fixes issues identified during mpi-pytest integration in sklearnex

### CODE REVIEW

File: cpp/oneapi/dal/algo/knn/backend/gpu/infer_kernel_impl_dpc_distr.hpp [hunk 1]
```diff
@@ -371,16 +365,10 @@ class knn_callback_distr {
 
         const auto& [first, last] = bnds;
         ONEDAL_ASSERT(last > first);
-        auto& queue = this->queue_;
-
-        bk::event_vector ndeps{ deps.cbegin(), deps.cend() };
-        auto sq_event = copy_with_sqrt(queue, inp_dts, inp_dts, deps);
```
Comment (Alexandr-Solovev) [inline cpp/oneapi/dal/algo/knn/backend/gpu/infer_kernel_impl_dpc_distr.hpp:377]: Can you clarify please how you avoid the usage of copy_with_sqrt func?

Comment (ethanglaser) [reply cpp/oneapi/dal/algo/knn/backend/gpu/infer_kernel_impl_dpc_distr.hpp:377]: It's done here already - https://github.com/oneapi-src/oneDAL/pull/2733/files#diff-74259405865e71161423cc6a67bfeef97953a7b9adc8ec21ef8f0162fa1a9d13L283

File: cpp/oneapi/dal/algo/knn/backend/gpu/infer_kernel_impl_dpc.hpp [hunk 2]
```diff
@@ -286,12 +286,12 @@ class knn_callback {
...
         ONEDAL_ASSERT(last > first);
-        auto& queue = this->queue_;
 
         bk::event_vector ndeps{ deps.cbegin(), deps.cend() };
-        auto sq_event = copy_with_sqrt(queue, inp_dts, inp_dts, deps);
-        if (this->compute_sqrt_)
+        if (this->compute_sqrt_) {
+            auto sq_event = copy_with_sqrt(this->queue_, inp_dts, inp_dts, deps);
             ndeps.push_back(sq_event);
+        }
```
Comment (ethanglaser) [inline cpp/oneapi/dal/algo/knn/backend/gpu/infer_kernel_impl_dpc.hpp:294]:
```suggestion
            auto sqrt_event = copy_with_sqrt(this->queue_, inp_dts, inp_dts, deps);
            ndeps.push_back(sqrt_event);
        }
```

File: cpp/oneapi/dal/algo/knn/backend/gpu/infer_kernel_impl_dpc.hpp [hunk 3]
```diff
@@ -310,12 +310,12 @@ class knn_callback {
...
```
Comment (ethanglaser) [inline cpp/oneapi/dal/algo/knn/backend/gpu/infer_kernel_impl_dpc.hpp:318]:
```suggestion
            auto sqrt_event = copy_with_sqrt(this->queue_, inp_dts, inp_dts, deps);
            ndeps.push_back(sqrt_event);
        }
```

**Review by Alexandr-Solovev — APPROVED** [review]

**Pull request merged**